File: Basic_problems/RomanToInteger.java

// Problem link : https://leetcode.com/problems/roman-to-integer/description/

package Basic_problems;

import java.util.*;

public class RomanToInteger {
    public static int romanToInt(String s) {
        Map<Character, Integer> romanMap = new HashMap<>();
        romanMap.put('I', 1);
        romanMap.put('V', 5);
        romanMap.put('X', 10);
        romanMap.put('L', 50);
        romanMap.put('C', 100);
        romanMap.put('D', 500);
        romanMap.put('M', 1000);

        int res = 0;
        for (int i = 0; i < s.length(); i++) {
            if (i < s.length() - 1 && romanMap.get(s.charAt(i)) < romanMap.get(s.charAt(i + 1))) {
                res -= romanMap.get(s.charAt(i));
            } else {
                res += romanMap.get(s.charAt(i));
            }
        }
        return res;
    }

    public static void main(String[] args) {
        String str = "III";

        // String str = "LVIII";

        // String str = "MCMXCIV";
        
        System.out.println(romanToInt(str));
    }
}

File: Basic_problems/RomanToInteger.js

// Problem link : https://leetcode.com/problems/roman-to-integer/description/

let romanToInt = function(s) {
    const romanMap = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
    };

    let res = 0;
    for(let i=0; i<s.length; i++){
        if(i < s.length-1 && romanMap[s[i]] < romanMap[s[i+1]]){
            res -= romanMap[s[i]];
        }else{
            res += romanMap[s[i]];
        }
    }
    return res;
};

// let str = "III";

// let str = "LVIII";

let str = "MCMXCIV";
console.log(romanToInt(str));
